ddcz/models/used/discussion.py

```diff
@@ -29,3 +29,22 @@ def user_profile_url(self):
     @property
     def by_registered_user(self):
         return self.reg == "1" and self.user
+
+    def createNewCommentFromForm(self, form, user, reputation=0):
+        self.nickname = user.nick_uzivatele
+        self.email = user.email_uzivatele
+        self.text = form.cleaned_data["text"]
+        self.reg = 1
+        self.reputace = reputation
+        self.user = user
+        self.save()
+
+    def deleteComment(self, comment_id):
+        if self.canBeDeleted(comment_id):
+            self.objects.get(id=comment_id).delete()
+
+    def canBeDeleted(self, comment_id):
```

Comment:
This is misleading and is the reason why the guards should be in the controller.

It can also be deleted by admin, for example, or by associated objects. If anything, `canBeDeletedByUser`

Reply:
also, I don't get it. Why do you have to provide comment_id and not do just `comment.delete()`?

Reply:
Also this seems to be dead code not used from view anyway...is this ready to merge or am I reviewing something that's WIP? If it is, please use the WIP tag :)